stream-compositions/services/product-composition-service/src/main/java/com/backbase/stream/compositions/product/core/service/impl/ProductIngestionServiceImpl.java

```diff
@@ -43,6 +44,7 @@ public class ProductIngestionServiceImpl implements ProductIngestionService {
      */
     public Mono<ProductIngestResponse> ingestPull(ProductIngestPullRequest ingestPullRequest) {
         return pullProductGroup(ingestPullRequest)
+                .map(response -> this.setServiceAgreementIds(ingestPullRequest, response))
```

Comment:
@piotr-backbase Could you move this to ProductIntegrationServiceImpl.pullProductGroup()? We already transform the integration response to the Stream model there.

Reply:
Definitely. It should be there. Done